include/gl_layer/private/context.h

```diff
@@ -22,13 +22,18 @@ class Context {
     void glGetProgramiv(unsigned int handle, GLenum param, int* params);
     void glLinkProgram(GLuint program);
     void glUseProgram(unsigned int handle);
+    void glDeleteProgram(GLuint program);
+
+    void validate_program_bound(std::string_view func_name);
+    bool validate_program_status(GLuint program);
 
 private:
     Version gl_version;
 
     GLLayerOutputFun output_fun = nullptr;
     void* output_user_data = nullptr;
     ContextGLFunctions gl;
+    GLuint bound_program = 0;
```

Comment:
would "current_program" make more sense, name wise?
technically for all intercepted values "current_" should be the prefix for all the actual gl-state?

Reply:
I think "bound" implies that it's set by `glUseProgram` pretty well.

Reply:
I like "bound", as it would also apply to `glBind*` functions.